src/engraving/dom/cmd.cpp

```diff
@@ -1752,14 +1751,6 @@ void Score::changeCRlen(ChordRest* cr, const Fraction& dstF, bool fillWithRest)
                 }
             }
         }
-        Measure* m  = cr1->measure();
-        Measure* m1 = m->nextMeasure();
-        if (m1 == 0) {
-            break;
-        }
-        Segment* s = m1->first(SegmentType::ChordRest);
-        expandVoice(s, track);
```

Comment:
Isn't it just the call to `expandVoice` that should be removed? Judging from what `flist` seems to be, it does seem crucial to move on to the next measure.

Reply:
From what I can tell, moving to the next measure functions the same as it has previously. Maybe I am missing something?

Reply:
Sorry, my comment was a bit ambiguous. I meant: it seems important that each iteration of this loop uses a ChordRest/Segment from the next measure. 

With "moving to the next measure", I was referring to this code:
```
        Measure* m  = cr1->measure();
        Measure* m1 = m->nextMeasure();
        if (m1 == 0) {
            break;
        }
        Segment* s = m1->first(SegmentType::ChordRest);
        …
        cr1 = toChordRest(s->element(track));
```
which ensured that cr1 is from a different measure on each iteration of the loop.

Reply:
It seems that this can lead to a corruption:
1. Create this situation:
    <img width="201" alt="Scherm­afbeelding 2024-02-12 om 03 34 03" src="https://github.com/musescore/MuseScore/assets/48658420/c81b79e3-23a1-4a62-8ae4-d65e01fe9d16">
    (second measure is copy of first, with the first two rests deleted)
2. Select the last rest of the first measure, and press '7' to turn that into a whole rest.
    The second measure gets corrupted:
    <img width="230" alt="Scherm­afbeelding 2024-02-12 om 03 37 41" src="https://github.com/musescore/MuseScore/assets/48658420/c9c704a9-042a-4562-a795-803796652c2d">
    Corruption info: "Voice too long: voice 2. Found: 5/4. Expected: 4/4."
    (weirder cases can be created by using different time signatures for example, like 3/8)